src/main/java/io/codiga/plugins/jetbrains/completion/inline/InlineDocumentListener.java

```diff
@@ -37,13 +37,13 @@ public class InlineDocumentListener implements DocumentListener {
     private static final int TIMEOUT_REQUEST_POLLING_MILLISECONDS = 500;
     private final DependencyManagement dependencyManagement = new DependencyManagement();
     private final Alarm updateListAlarm = new Alarm();
+    private final CodigaApi codigaApi = CodigaApi.getInstance();
+    private final AppSettingsState settings = AppSettingsState.getInstance();
```

Comment:
That should not be at the top. This is something Yann from JetBrains told me : by requesting services in the main class, it increases load time and service should be requested when we need them, not when we start the plugin. Doing so decrease the overall IDE experience.

Reply:
I removed the fields, and moved the service initialization calls to the locations where they are actually used.